### PR TITLE
Import tlz

### PR DESCRIPTION
**Pull request by jakirkham**
Same change as PR ( https://github.com/dask/dask/pull/5997 ) except for Distributed.

cc @jrbourbeau

### CODE REVIEW

File: distributed/dashboard/components/scheduler.py [hunk 1]
```diff
@@ -35,7 +35,8 @@
 from bokeh.io import curdoc
 import dask
 from dask.utils import format_bytes, key_split
-from toolz import pipe
+from tlz import pipe
+from tlz.curried import map, concat, groupby
```
Comment (jrbourbeau) [inline distributed/dashboard/components/scheduler.py]: Can use built in `map` function

File: distributed/protocol/core.py [hunk 2]
```diff
@@ -3,10 +3,7 @@
 
 import msgpack
 
-try:
-    from cytoolz import reduce
-except ImportError:
-    from toolz import reduce
+from tlz import reduce
```
Comment (jrbourbeau) [inline distributed/protocol/core.py]: Can use `reduce` from `functools`

File: distributed/tests/test_client.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 import zipfile
 
 import pytest
-from toolz import identity, isdistinct, concat, pluck, valmap, partial, first, merge
+from tlz import identity, isdistinct, concat, pluck, valmap, partial, first, merge
```
Comment (jrbourbeau) [inline distributed/tests/test_client.py]: Can use `partial` from `functools`

File: distributed/worker.py [hunk 4]
```diff
@@ -21,10 +21,7 @@
 from dask.utils import format_bytes, funcname
 from dask.system import CPU_COUNT
 
-try:
-    from cytoolz import pluck, partial, merge, first, keymap
-except ImportError:
-    from toolz import pluck, partial, merge, first, keymap
+from tlz import pluck, partial, merge, first, keymap
```
Comment (jrbourbeau) [inline distributed/worker.py]: Can use `partial` from `functools`

File: distributed/dashboard/components/scheduler.py (hunk 1 above)
Comment (jakirkham) [reply distributed/dashboard/components/scheduler.py]: This is curried so idk if that would work here.

Comment (jrbourbeau) [reply distributed/dashboard/components/scheduler.py]: Ah, yep you're right. I overlooked the `curried` part of this `import`

**Comment by jakirkham** [comment]
Thanks James! Have updated this to use `functools` where possible. Would appreciate another look when you have a chance 🙂

**Comment by jrbourbeau** [comment]
Thanks @jakirkham!

**Comment by jakirkham** [comment]
Thanks for the review 🙂

**Pull request merged**